### ISSUE TITLE
Check that client state is zeroed out for IBC client upgrade proposals

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#291]
## Summary

proposal.ValidateBasic() should ensure the client submitted is zeroed out. Not a security concern but prevents the proposal from containing information governance is not actually voting on

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by alizahidraja** [comment]
Hey!
I wanted to ask, in what cases is the client state not needed? And also want to confirm that we’re talking about the client state associated with UpgradeProposal, right?

**Comment by colin-axner** [comment]
> Hey! I wanted to ask, in what cases is the client state not needed? And also want to confirm that we’re talking about the client state associated with UpgradeProposal, right?

Yes we are talking about the client state in the upgrade proposal. The client is always needed. "Zeroing" out the client state just means removing all relayer set fields from the client state (setting them to nil/zero)

**Comment by crodriguezvega** [comment]
> No, we don't want to add another function to the exported.ClientState. We can simply check that reflect.DeepEqual(up.UpgradedClientState, up.UpgradedClientState.ZeroCustomFields()) ie that the custom fields have already been zeroed out on the provided upgraded client state

I think we need to first [unpack](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/proposal.go#L114), right? So then we just need to do something like this?

```
clientState, err := UnpackClientState(up.UpgradedClientState)
if err != nil {
        return sdkerrors.Wrap(err, "failed to unpack upgraded client state")
}

if !reflect.DeepEqual(clientState, clientState.ZeroCustomFields()) {
	return sdkerrors.Wrap(err, "upgraded client state is not zeroed out")
}
```

**Comment by colin-axner** [comment]
> I think we need to first [unpack](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/proposal.go#L114), right?

Yup, but the unpacking is already being performed in `proposal.ValidateBasic()`

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
